push-to-s3-and-invalidate-cloudfront/action.yml

```diff
@@ -0,0 +1,150 @@
+name: Push files to an S3 bucket destination and invalidate the cloudfront cache
+description: |
+  This action will push files to a destination S3 bucket (or a folder within a bucket) and invalidates the cloudfront cache.
+  The action also removes any stale files from the bucket. The action can also delete files/folders from the bucket.
+
+inputs:
+  build-directory:
+    description: The directory that contains the build files to upload to S3
+    required: true
+  pull-request-number:
+    description: The pull request number for which a preview needs to be generated/updated
+    required: true
```

Comment:
Is it correct to have this as required? If we are calling this from a workflow triggered `on.push`, then there will be no pull-request-number.

Reply:
Good catch! We only need it if the `s3-directory` variable is to be derived from the context. Made it `required: false` and added a step to check for this if the `s3-directory` input is not set